### ISSUE TITLE
Assert failure: HasPerInstInfo()

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#25244]
**Job:**
https://mc.dot.net/#/user/coreclr-outerloop-gcstress-extra/ci~2Fdotnet~2Fcoreclr~2Frefs~2Fheads~2Fmaster/test~2Ffunctional~2Fcli~2F/20190616.1/workItem/JIT/analysis/xunit/JIT_Performance._CodeQuality_BenchmarksGame_regex_redux_regex_redux_5_regex_redux_5_~2F_CodeQuality_BenchmarksGame_regex_redux_regex_redux_5_regex_redux_5_cmd

**Failed tests:**
JIT_Performance._CodeQuality_BenchmarksGame_regex_redux_regex_redux_5_regex_redux_5_._CodeQuality_BenchmarksGame_regex_redux_regex_redux_5_regex_redux_5_cmd

**Log:**
```
Assert failure(PID 4460 [0x0000116c], Thread: 3048 [0x0be8]): HasPerInstInfo()

<no module>! <no symbol> + 0x0 (0x00000000)
    File: f:\\workspace\\_work\\1\\s\\src\\vm\\methodtable.h Line: 2955
    Image: C:\\dotnetbuild\\work\\d612e967-3739-4ea0-bbcf-da077764d85b\\Payload\\CoreRun.exe


Return code:      1
Raw output file:      C:\\dotnetbuild\\work\\d612e967-3739-4ea0-bbcf-da077764d85b\\Work\\e39edb3b-c408-480c-a752-84aca6598fdc\\Exec\\JIT\\Performance\\Reports\\JIT.Performance\\CodeQuality\\BenchmarksGame\\regex-redux\\regex-redux-5\\regex-redux-5.output.txt
Raw output:
BEGIN EXECUTION
 \"C:\\dotnetbuild\\work\\d612e967-3739-4ea0-bbcf-da077764d85b\\Payload\\corerun.exe\" regex-redux-5.exe 
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\\dotnetbuild\\work\\d612e967-3739-4ea0-bbcf-da077764d85b\\Payload
> C:\\dotnetbuild\\work\\d612e967-3739-4ea0-bbcf-da077764d85b\\Work\\e39edb3b-c408-480c-a752-84aca6598fdc\\Exec\\JIT\\Performance\\CodeQuality\\BenchmarksGame\\regex-redux\\regex-redux-5\\regex-redux-5.cmd
Expected: True
Actual:   False
```

**Comment by RussKeldorph** [comment]
regex-redux-5 has failed 16 times under GCStress 
```
TestResults
| where Result != "Pass" and Message !contains ("coreclr_initialize") and Message !contains("disassembler") and Method contains("regex_redux")
| join (Jobs | where Source contains("coreclr/refs/heads/master")) on JobId
| project Started, QueueName, Properties, Duration, Message
```

**Comment by RussKeldorph** [comment]
Almost all of the ARM/ARM64 failures look like #25049, but the x86 messages are this issue (`HasPerInstInfo`) and the following:
```
!CREATE_CHECK_STRING(!\"Detected use of a corrupted OBJECTREF. Possible GC hole.\")
```
and
```
Consistency check failed: AV in clr at this callstack:
```

**Comment by RussKeldorph** [comment]
@dotnet/jit-contrib

**Comment by RussKeldorph** [comment]
Sorry, I confused myself.  The ARM failures look like #25048.  One x86 failure looks like #25049.

**Comment by briansull** [comment]
I ran `GCSTRESS=C` in a loop on
`JIT\Performance\CodeQuality\BenchmarksGame\regex-redux\regex-redux-5\regex-redux-5.exe`

And saw one failure that looked like a GC hole:

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Reflection.Emit.ILGenerator.Emit(OpCode opcode, Label label)
   at System.Text.RegularExpressions.RegexCompiler.BleFar(Label l) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 548
   at System.Text.RegularExpressions.RegexCompiler.GenerateFindFirstChar() in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 1465
   at System.Text.RegularExpressions.RegexLWCGCompiler.FactoryInstanceFromCode(RegexCode code, RegexOptions options) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexLWCGCompiler.cs:line 39
   at System.Text.RegularExpressions.RegexCompiler.Compile(RegexCode code, RegexOptions options) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 171
   at System.Text.RegularExpressions.Regex.Compile(RegexCode code, RegexOptions roptions) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs:line 213
   at System.Text.RegularExpressions.Regex..ctor(String pattern, RegexOptions options, TimeSpan matchTimeout, Boolean addToCache) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs:line 160
   at System.Text.RegularExpressions.Regex..ctor(String pattern, RegexOptions options) in /_/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs:line 61

```

**Comment by briansull** [comment]
Likely that this issue is fixed, but still waiting on the testing infrastructure to confirm this

**Comment by briansull** [comment]
Resolved by #25445
